examples/vqe.lisp

```diff
@@ -0,0 +1,235 @@
+;;;; examples/vqe.lisp
+;;;;
+;;;; Author: Juan M. Bello-Rivas
+
+(in-package #:qvm-examples)
+
+;;; Example of the usage of the variational quantum eigensolver algorithm to
+;;; compute the ground state energy of an H2 molecule.  The hard-coded values
+;;; below are for a conformation where the inter-atomic distance is 0.5
+;;; Angstroms. The reference value for the ground energy at this conformation
+;;; is -1.05515979 Hartree.
+
+(defparameter *number-of-qubits* 4
+  "Number of qubits in the quantum virtual machine")
+
+(defparameter *reference-state* "X 0; X 1" "Reference state program (HF).")
+
+(defparameter *ansatz* "
```

Comment:
Is there a paper you could reference here for interested readers? How is the below ansatz chosen?

Reply:
Just added a comment about it.